### PR TITLE
Allow users of nunit3 set environment variables in the runner

### PR DESCRIPTION
**Pull request by robertpi**
I need this for an obscure reason, maybe other people will too?

### CODE REVIEW

**Comment by matthid** [comment]
Hey, thanks for sending in a PR :)
To be honest I'm a bit skeptical, because when this is needed people should fall back to the lower level APIs (ideally we return the `CreateProcess` instance somewhere). I quickly tried to find some other place we have this and found none. Do we have this somewhere else (Because if we decide to add this it should work similar to existing APIs)?

**Comment by robertpi** [comment]
Seems the Fsi module is the thing to copy, there's some clever use of inlining in `Process.setEnvironmentVariable` that allows it operate on any type that has a `Environment` and `WithEnvironment` member.

I think it should be easy to change the PR to copy that pattern.

**Comment by robertpi** [comment]
As the commit message says, we now do exactly what other task such as DotNet and Fsi do.

File: src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs [hunk 1]
```diff
@@ -330,7 +338,7 @@ let internal createProcess createTempFile (setParams : NUnit3Params -> NUnit3Par
     let argLine = Args.toWindowsCommandLine [ (sprintf "@%s" path) ]
     CreateProcess.fromRawCommandLine tool argLine
     |> CreateProcess.withFramework
-    |> CreateProcess.withWorkingDirectory (getWorkingDir parameters)
+    |> CreateProcess.withEnvironment  (parameters.Environment |> Map.toList)
```
Comment (matthid) [inline src/app/Fake.DotNet.Testing.NUnit/NUnit3.fs:341]: We sill want to set the working directory, correct?

**Comment by matthid** [comment]
Thanks! Will probably take me a bit to release, but it will be part of the next one.

**Pull request merged**